src/discordtipbot/discordbot.cr

```diff
@@ -133,6 +121,38 @@ class DiscordBot
       end
     end
 
+    # Handle new guilds and owner notifying etc
+    @streaming = false
+
+    @bot.on_ready do |payload|
+      # Only fire on first READY, or if ID cache was cleared
+      next unless @unavailable_guilds.empty?
+      @streaming = true
+      @unavailable_guilds.concat payload.guilds.map(&.id)
+    end
+
+    @bot.on_guild_create do |payload|
+      if @streaming
+        @available_guilds.add payload.id
+
+        # Done streaming
+        if @available_guilds == @unavailable_guilds
+          # Process guilds at a rate
+          @cache.guilds.each do |guild|
```

Comment:
You can make this more clear by breaking out the tuple in the block args https://carc.in/#/r/3jyc, i.e. `|_id, guild|`